src/fragment-components/a-radio-group.tsx

```diff
@@ -0,0 +1,207 @@
+	return (
+		<AComponent
+		componentObj={componentObj}
+		{...rest}>
+			<RadioButtonGroup
+				className={componentObj.cssClasses?.map((cc: any) => cc.id).join(' ')}
+				legendText= {componentObj.legend}
+				disabled= {componentObj.disabled}
+				orientation={componentObj.orientation}
+				defaultSelected={componentObj.defaultSelected}
+				labelPosition={componentObj.labelPosition}
+				name={componentObj.codeContext?.name}>
+					{children}
+			</RadioButtonGroup>
+		</AComponent>
+	);
+};
+
+export const componentInfo: ComponentInfo = {
+	component: ARadioButtonGroup,
+	styleUI: ARadioButtonGroupStyleUI,
+	codeUI: ARadioButtonGroupCodeUI,
+	keywords: ['radio', 'button', 'group'],
+	name: 'Radio buttons',
+	defaultComponentObj: {
+		disabled: false,
+		type: 'radioGroup',
+		legend: 'Radio Button Group',
+		defaultSelected: '',
+		labelPosition: 'right',
+		orientation: 'horizontal',
+		items: [
+			{
+				type: 'radio',
+				labelText: 'Option 1',
+				disabled: false,
+				defaultChecked: true
+			},
+			{
+				type: 'radio',
+				labelText: 'Option 2',
+				disabled: false,
+				defaultChecked: false
+			},
+			{
+				type: 'radio',
+				labelText: 'Option 3',
+				disabled: false,
+				defaultChecked: false
+			}
+		]
+	},
+	render: ({ componentObj, select, remove, selected, renderComponents }) => <ARadioButtonGroup
+		componentObj={componentObj}
+		select={select}
+		remove={remove}
+		selected={selected}>
+			{componentObj.items.map((button: any) => (renderComponents(button)))}
+	</ARadioButtonGroup>,
+	image,
+	codeExport: {
+		angular: {
+			inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}LegendText = "${json.legend}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Orientation = "${json.orientation}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}LabelPosition = "${json.labelPosition}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Name = "${json.codeContext?.name}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}defaultSelected = "${json.defaultSelected}";`,
+			outputs: ({ json }) => `@Output() ${nameStringToVariableString(json.codeContext?.name)}ValueChange = new EventEmitter();`,
+			imports: ['RadioModule'],
+			code: ({ json, jsonToTemplate }) => {
+				return `<legend class="bx--label">{{${nameStringToVariableString(json.codeContext?.name)}LegendText}}</legend>
+				<ibm-radio-group
+					[name]="${nameStringToVariableString(json.codeContext?.name)}Name"
+					[orientation]="${nameStringToVariableString(json.codeContext?.name)}Orientation"
+					[labelPlacement]="${nameStringToVariableString(json.codeContext?.name)}LabelPosition"
+					(change)="${nameStringToVariableString(json.codeContext?.name)}ValueChange.emit($event.value)"
+					${angularClassNamesFromComponentObj(json)}>
+						${json.items.map((element: any) => jsonToTemplate(element)).join('\n')}
+				</ibm-radio-group>`;
```

Comment:
Looks like there's no way to set initial checked value? default selected isn't used anywhere?
Would it make sense to use ngModel instead (or in addition)?

Reply:
you can set the default selected value from the dropdown default selected. that sets what checkbox should be initially checked in the radio group items.

Reply:
right, that's the UI, but where is that used in angular code export?

Reply:
in the json.items, each item has a checked and default selected value.